src/cbmc/cbmc_parse_options.cpp

```diff
@@ -515,6 +515,10 @@ int cbmc_parse_optionst::doit()
   if(get_goto_program_ret!=-1)
     return get_goto_program_ret;
 
+  INVARIANT(
+    !goto_model.check_internal_invariants(*this),
```

Comment:
`check_consistency`?

Reply:
I find these a little bit confusing in design, both here and in #3118. Do you think it would be better if `check_internal_invariants` returned `true` by default, as the *all is good* case, and `false` otherwise? This invariant would also look cleaner as well, as it would just read as `the internal invariants of the goto model should always hold, which this is what the invariant will check by essence of it returning true` and not `if this doesn't return false, then the internal invariants hold`. Like, it's the double negative both in the condition here, and the return values of `check_internal_invariants` that's slightly confusing, as it's inverse logic to what's expected. But that's just me, certainly not required.

Reply:
Within this PR, yes. But we are preparing a larger collection of checks and not all of them target consistency.

Reply:
@NlightNFotis it was my understanding that we prefer returning `true` when things fail. It can of course be negated.

Reply:
@NlightNFotis My understanding is that linux programs return 0 to indicate no error and a non-zero integer to represent some kind of error. C programs copy this and return 0 to indicate no error. We follow this by returning `false` to indicate no error and `true` to indicate error. I find it confusing too.